apps/test/unit/code-studio/scriptEditorReduxTest.js

```diff
@@ -52,7 +68,25 @@ describe('scriptEditorRedux reducer tests', () => {
     const nextState = reducer(initialState, addGroup(2, 'key', 'Display Name'))
       .lessonGroups;
     assert.equal(nextState[nextState.length - 1].display_name, 'Display Name');
+    assert.equal(nextState[nextState.length - 1].user_facing, true);
   });
+
+  it('remove group', () => {
+    // Remove lesson group when there are 2 lessons groups
+    let nextState = reducer(initialState, removeGroup(1));
+    let lessonGroups = nextState.lessonGroups;
+    assert.equal(lessonGroups.length, 1);
+    assert.equal(lessonGroups[0].key, 'lg-key-2');
+
+    // Remove lesson group when there is only one lesson group left
+    // a non-user facing lesson group should be added
```

Comment:
If I am understanding correctly then I'm a bit worried about this user flow - if i enable lesson groups, then realize I've made a mistake, it seems much more likely that I'll want to convert the one remaining lesson group back to non-user-facing, as opposed to deleting all my lessons. I don't have an exact solution in mind, but accidentally clicking "enable" seems like a bit of a land mine for curriculum writers.

Reply:
I'll look at how hard it is to have a disable lesson groups button if there is only one lesson group

Reply:
ok, this would be fine to resolve in a future PR / discussion if that's easier.